docs/Guides/Testing.md

```diff
@@ -243,33 +243,68 @@ after initializing routes and plugins with `fastify.ready()`.
 
 Uses **app.js** from the previous example.
 
-**test-listen.js** (testing with
-[`Request`](https://www.npmjs.com/package/request))
+**test-listen.js** (testing with [`undici`](https://www.npmjs.com/package/undici))
 ```js
 const tap = require('tap')
-const request = require('request')
+const { Client } = require('undici')
 const buildFastify = require('./app')
 
-tap.test('GET `/` route', t => {
-  t.plan(5)
+tap.test('should work with undici', async t => {
+  t.plan(2)
 
   const fastify = buildFastify()
 
-  t.teardown(() => fastify.close())
+  await fastify.listen()
 
-  fastify.listen({ port: 0 }, (err) => {
-    t.error(err)
+   const client = new Client(
+    'http://localhost:' + fastify.server.address().port, {
+      keepAliveTimeout: 10,
+      keepAliveMaxTimeout: 10
```

Comment:
it may be worth pointing out why these are needed. I think there's a section in undici's docs about this right?

Reply:
Yes, here is the `undici` documentation:  https://undici.nodejs.org/#/docs/api/Client